### PR TITLE
chore: allow updating stage0 via workflow_dispatch

### PR DESCRIPTION
**Pull request by nomeata**
follow-up to #3042

### CODE REVIEW

File: .github/workflows/update-stage0.yml [hunk 1]
```diff
@@ -23,16 +24,17 @@ jobs:
     - uses: actions/checkout@v3
       with:
         ssh-key: ${{secrets.STAGE0_SSH_KEY}}
-    - name: Check if update is needed
+    - run: echo "DOIT=yes" >> "$GITHUB_ENV"
```
Comment (semorrison) [inline .github/workflows/update-stage0.yml:27]: I'm not super excited that the flag is named `DOIT` here!

**Comment by semorrison** [comment]
Could we have a paragraph *somewhere* (not quite sure the best place? dev guide in the manual?) explaining how to invoke this via `workflow_dispatch` manually?

File: .github/workflows/update-stage0.yml (hunk 1 above)
Comment (nomeata) [reply .github/workflows/update-stage0.yml:27]: Happy to change, any suggestions?

**Comment by nomeata** [comment]
> Could we have a paragraph _somewhere_ (not quite sure the best place? dev guide in the manual?) explaining how to invoke this via `workflow_dispatch` manually?

Good idea, will do. The bootstrapping section of the manual probably

File: .github/workflows/update-stage0.yml (hunk 1 above)
Comment (nomeata) [reply .github/workflows/update-stage0.yml:27]: Is `should_update_stage0` better?

**Comment by nomeata** [comment]
I changed the flag, and also rewrote the section on updating stage0 in the manual, PTAL

**Review by Kha — APPROVED** [review]

**Review by semorrison — APPROVED** [review]

File: .github/workflows/update-stage0.yml (hunk 1 above)
Comment (semorrison) [reply .github/workflows/update-stage0.yml]: Just unclear. (I have a vendetta against 'it'.)

**Pull request merged**